braid/relax.c

```diff
@@ -60,17 +72,65 @@ _braid_FCRelax(braid_Core  core,
          }
 
          /* F-relaxation */
+         _braid_GetRNorm(core, -1, &rnm);
          for (fi = flo; fi <= fhi; fi++)
          {
             _braid_Step(core, level, fi, NULL, u);
             _braid_USetVector(core, level, fi, u, 0);
+
+            /* Allow user to process current vector */
+            if( (access_level >= 3) || (done == 1) )
+            {
+               _braid_AccessStatusInit(ta[fi-f_ilower], fi, rnm, iter, level, nrefine, gupper,
+                                       done, 0, braid_ASCaller_FCRelax, astatus);
+               _braid_AccessVector(core, astatus, u);
+            }
+
+            /* If braid is finished, store last time step */
+            if (done && fi == ntime && level == 0 && storage < 0)
+            {
+               if (_braid_GridElt(grids[level], ulast) != NULL)
+               {
+                  _braid_BaseFree(core, app, _braid_GridElt(grids[level], ulast));
+                  _braid_GridElt(grids[level], ulast) = NULL;
+               }
+               _braid_BaseClone(core, app,  u, &(_braid_GridElt(grids[level], ulast)));
+            }
+
          }
 
          /* C-relaxation */
          if (ci > _braid_CoreElt(core, initiali))
          {
             _braid_Step(core, level, ci, NULL, u);
             _braid_USetVector(core, level, ci, u, 1);
+            
+            /* Allow user to process current vector, noting that if done */
+            if( (access_level >= 3) || (done == 1) )
+            {
+               _braid_AccessStatusInit(ta[ci-f_ilower], ci, rnm, iter, level, nrefine, gupper,
+                                       done, 0, braid_ASCaller_FCRelax, astatus);
+               _braid_AccessVector(core, astatus, u);
+            }
+         
+            /* If braid is finished, store last time step */
+            if (done && ci == ntime && level == 0 && storage < 0)
+            {
+               if (_braid_GridElt(grids[level], ulast) != NULL)
+               {
+                  _braid_BaseFree(core, app, _braid_GridElt(grids[level], ulast));
+                  _braid_GridElt(grids[level], ulast) = NULL;
+               }
+               _braid_BaseClone(core, app,  u, &(_braid_GridElt(grids[level], ulast)));
+            }
+         }
+         else if( (level == 0) && (ci == _braid_CoreElt(core, initiali)) && done)
```

Comment:
@rfalgout @steffi7574

Here, you can see one of the new User-Access hooks FCRelax needed.   This is the hook to make sure access is given at time t==0, if Braid is done.  The other new User-Access hooks are above.